homeassistant/components/zodiac/sensor.py

```diff
@@ -0,0 +1,221 @@
+            ATTR_MODALITY: MODALITY_MUTABLE,
+        },
+    ),
+    (
+        (22, 9),
+        (22, 10),
+        {
+            ATTR_SIGN: SIGN_LIBRA,
+            ATTR_ELEMENT: ELEMENT_AIR,
+            ATTR_MODALITY: MODALITY_CARDINAL,
+        },
+    ),
+    (
+        (23, 10),
+        (22, 11),
+        {
+            ATTR_SIGN: SIGN_SCORPIO,
+            ATTR_ELEMENT: ELEMENT_WATER,
+            ATTR_MODALITY: MODALITY_FIXED,
+        },
+    ),
+    (
+        (23, 11),
+        (21, 12),
+        {
+            ATTR_SIGN: SIGN_SAGITTARIUS,
+            ATTR_ELEMENT: ELEMENT_FIRE,
+            ATTR_MODALITY: MODALITY_MUTABLE,
+        },
+    ),
+    (
+        (22, 12),
+        (20, 1),
+        {
+            ATTR_SIGN: SIGN_CAPRICORN,
+            ATTR_ELEMENT: ELEMENT_EARTH,
+            ATTR_MODALITY: MODALITY_CARDINAL,
+        },
+    ),
+    (
+        (21, 1),
+        (19, 2),
+        {
+            ATTR_SIGN: SIGN_AQUARIUS,
+            ATTR_ELEMENT: ELEMENT_AIR,
+            ATTR_MODALITY: MODALITY_FIXED,
+        },
+    ),
+    (
+        (20, 2),
+        (20, 3),
+        {
+            ATTR_SIGN: SIGN_PISCES,
+            ATTR_ELEMENT: ELEMENT_WATER,
+            ATTR_MODALITY: MODALITY_MUTABLE,
+        },
+    ),
+)
+
+ZODIAC_ICONS = {
+    SIGN_ARIES: "mdi:zodiac-aries",
+    SIGN_TAURUS: "mdi:zodiac-taurus",
+    SIGN_GEMINI: "mdi:zodiac-gemini",
+    SIGN_CANCER: "mdi:zodiac-cancer",
+    SIGN_LEO: "mdi:zodiac-leo",
+    SIGN_VIRGO: "mdi:zodiac-virgo",
+    SIGN_LIBRA: "mdi:zodiac-libra",
+    SIGN_SCORPIO: "mdi:zodiac-scorpio",
+    SIGN_SAGITTARIUS: "mdi:zodiac-sagittarius",
+    SIGN_CAPRICORN: "mdi:zodiac-capricorn",
+    SIGN_AQUARIUS: "mdi:zodiac-aquarius",
+    SIGN_PISCES: "mdi:zodiac-pisces",
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Zodiac sensor platform."""
+    if discovery_info is None:
+        return
+
+    async_add_entities([ZodiacSensor()], True)
+
+
+class ZodiacSensor(Entity):
+    """Representation of a Zodiac sensor."""
+
+    def __init__(self):
+        """Initialize the zodiac sensor."""
+        self._attrs = None
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return DOMAIN
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return "Zodiac"
+
+    @property
+    def device_class(self):
+        """Return the device class of the entity."""
+        return "zodiac__sign"
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._attrs[STATE_ATTR]
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ZODIAC_ICONS.get(self._attrs[STATE_ATTR])
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attrs
+
+    async def async_update(self):
```

Comment:
@MartinHjelmare could i ask a favor? do you see why this would take more than 30s to update? it keeps running the update over and over. timing out the CI

Reply:
I'm looking into it.

Reply:
It was a patch scope problem. The patch modified the dt util module instead of just the imported name in the zodiac sensor module. That tampered with the time keeping in the app and made some cycle.

Reply:
ah i almost wondered if if was something like that but saw other tests doing it so had discarded the thought

Reply:
appreciate the look thanks!